src/Airship/Internal/Route.hs

```diff
@@ -32,7 +42,25 @@ data BoundOrUnbound = Bound Text
 instance IsString Route where
     fromString s = Route [Bound (fromString s)]
 
-runRouter :: RoutingSpec m a -> [(Route, Resource m)]
+
+data RouteLeaf m = RouteMatch (Resource m) [Text]
+                 | RVar
+                 | RouteMatchOrVar (Resource m) [Text]
+                 | Wildcard (Resource m)
+
+instance Monad m => Monoid (RouteLeaf m) where
```

Comment:
This violates _all_ of the Monoid laws. Could we instead define a function to achieve the same matching?